### PR TITLE
Conda-forge recipe to add pydrift 0.2.1 to the channel

### PR DESCRIPTION
**Pull request by sergiocalde94**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by sergiocalde94** [comment]
Ready for review @conda-forge-admin

**Comment by sergiocalde94** [comment]
@conda-forge/staged-recipes This is ready for review and merge. Thanks!

**Comment by sergiocalde94** [comment]
@conda-forge-admin @conda-forge/core ?? I don't know how to ping :(

**Comment by sergiocalde94** [comment]
@luizirber I see you merged a PR some minutes ago, can you help me because I don´t know how to pin your team :(

**Comment by luizirber** [comment]
@sergiocalde94 You might want to send a message in https://gitter.im/conda-forge/conda-forge.github.io

File: recipes/pydrift/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  sha256: 572a5f5ee3cfde052ee858cc2ed741fbb5c3df9fbcd44a70fe1d208b751a163e
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
```
Comment (jan-janssen) [inline recipes/pydrift/meta.yaml:19]: You can remove this comment

File: recipes/pydrift/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - {{ compiler('c') }}
+  host:
+    - python
```
Comment (jan-janssen) [inline recipes/pydrift/meta.yaml:23]: For pip packages it is sufficient to have the packages listed in the run section.

File: recipes/pydrift/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    - jupyter
+    - typing-extensions
+    - catboost
+    - shap
+    - pre-commit
+    - pytest
+    - coveralls
+  run:
+    - python
+    - poetry
```
Comment (jan-janssen) [inline recipes/pydrift/meta.yaml:40]: According to the pip dependencies: 

```
- catboost >=0.23,<0.24
- coveralls >=2.0.0,<3.0.0
- flake8 >=3.8.1,<4.0.0
- jupyter >=1.0.0,<2.0.0
- missingno >=0.4.2,<0.5.0
- pandas >=1.0.3,<2.0.0
- plotly_express >=0.4.1,<0.5.0
- pre-commit >=2.4.0,<3.0.0
- pytest >=5.4.2,<6.0.0
- scikit-learn >=0.23.1,<0.24.0
- shap >=0.35.0,<0.36.0
- sphinx >=3.0.3,<4.0.0
- sphinx_press_theme >=0.5.1,<0.6.0
- typing-extensions >=3.7.4,<4.0.0
```

Comment (sergiocalde94) [reply recipes/pydrift/meta.yaml]: @jan-janssen I removed the comment and also removing all of dependencies from host, adapting the run ones to your message

**Comment by jan-janssen** [comment]
The package `sphinx_press_theme` is not available in conda yet

**Comment by sergiocalde94** [comment]
Removed pachake sphinx_press_theme and others that are not needed for pydrift (these libraries are only for developing part, sorry for including them before)

Thank you very much! @jan-janssen

**Comment by sergiocalde94** [comment]
I need to add poetry to host, if not it fail @jan-janssen

File: recipes/pydrift/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  sha256: 572a5f5ee3cfde052ee858cc2ed741fbb5c3df9fbcd44a70fe1d208b751a163e
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (chrisburr) [inline recipes/pydrift/meta.yaml:19]: I guess this isn't using a compiler?
```suggestion
```

File: recipes/pydrift/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    - pip
+    - poetry
+  run:
+    - python
+    - pip
+    - catboost >=0.23,<0.24
+    - jupyter >=1.0.0,<2.0.0
+    - pandas >=1.0.3,<2.0.0
+    - plotly_express >=0.4.1,<0.5.0
+    - pytest >=5.4.2,<6.0.0
```
Comment (chrisburr) [inline recipes/pydrift/meta.yaml:31]: Is `pytest` really a runtime requirement? Also do all of these version requirements need to be this strict?

Comment (sergiocalde94) [reply recipes/pydrift/meta.yaml:31]: I´ll remove pytest, you are right

The version requirements... they are the versions I ran, I don't know it the package will break if you install another version, I know for sure that there are a couple of warnings in previous catboost versions but I don't know more about this topic :(

Comment (chrisburr) [reply recipes/pydrift/meta.yaml]: Normally people only use a lower limit, i.e. `scikit-learn >=0.23.1`. Having an upper limit makes it:
* impossible to use this package with a newer version
* likely impossible to install other packages which have strict versions constraints like this

Normally upper limits are only used for compiled packages where the ABI can change. Most of the time this managed automatically using `run_exports` in packages that provide shared libraries and updates are managed using migrations (see the status and doc pages on conda-forge.org).

Comment (sergiocalde94) [reply recipes/pydrift/meta.yaml]: Youre absolutely right! Sorry I copied it from other proposal and I let it like that, now there is only lower limit in myu dependencies :)

File: recipes/pydrift/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    - pydrift
+
+about:
+  home: https://github.com/sergiocalde94/Data-And-Model-Drift-Checker
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'Measuring the degradation of a machine learning model'
+
+  # The remaining entries in this section are optional, but recommended.
```
Comment (marcelotrevisani) [inline recipes/pydrift/meta.yaml:44]:
```suggestion
```

File: recipes/pydrift/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+requirements:
+  host:
+    - python
+    - pip
+    - poetry
+  run:
+    - python
+    - pip
+    - catboost >=0.23
+    - jupyter >=1.0.0
```
Comment (marcelotrevisani) [inline recipes/pydrift/meta.yaml:26]: Could you please double-check this dependencies? I didn't see it on the metadata
```suggestion
```

File: recipes/pydrift/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    - pandas >=1.0.3
+    - plotly_express >=0.4.1
```
Comment (marcelotrevisani) [inline recipes/pydrift/meta.yaml:28]: And this one as well please
```suggestion
```

**Comment by marcelotrevisani** [comment]
If you want to, next try you can use [grayskull :skull: ](https://github.com/marcelotrevisani/grayskull/) to generate recipes for packages on ``pypi``

**Comment by sergiocalde94** [comment]
> If you want to, next try you can use [grayskull 💀 ](https://github.com/marcelotrevisani/grayskull/) to generate recipes for packages on `pypi`

I wish I'd heard about Celium sooner!! Thank you very much

**Comment by sergiocalde94** [comment]
@marcelotrevisani can you check it now? I think all things are now good :)

**Pull request merged**